checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileExporter.java

```diff
@@ -63,19 +67,39 @@ public void exportProfile(RulesProfile profile, Writer writer) {
             final List<ActiveRule> activeRules = profile
                     .getActiveRulesByRepository(CheckstyleConstants.REPOSITORY_KEY);
             if (activeRules != null) {
-                final Map<String, List<ActiveRule>> activeRulesByConfigKey =
-                        arrangeByConfigKey(activeRules);
+                final List<ActiveRuleWrapper> activeRuleWrappers = new ArrayList<>();
+                for (ActiveRule activeRule : activeRules) {
+                    activeRuleWrappers.add(new ActiveRuleWrapperImpl(activeRule));
+                }
+                final Map<String, List<ActiveRuleWrapper>> activeRulesByConfigKey =
+                        arrangeByConfigKey(activeRuleWrappers);
                 generateXml(writer, activeRulesByConfigKey);
             }
         }
         catch (IOException ex) {
             throw new IllegalStateException("Fail to export the profile " + profile, ex);
         }
+    }
+
+    protected void exportProfile(ActiveRules activeRules, Writer writer) {
```

Comment:
looks like you do `protected` for UT, it is not good.
I know there bunch code around that do the same, it is very bad practice, that is left from previous maintainers.

UT should be able to pass all branches through public entry points, in case where need to cover exceptional cases it is better to use reflection and explain in comment over test why reflection is the only way out.

Reply:
No, it's `protected` because of `https://github.com/checkstyle/sonar-checkstyle/pull/193/files#diff-515b093b908aa69201af7db1921c8263R77`. I could make it public as well just like the implemented abstract method but wanted to keep visibility as low as possible as it is kinda a "convenience method" imo to make use of the same exporter on the scanner side.

Reply:
please make it public.
We already have public method `exportProfile` with different arguments, lets have one more, especially if it suppose to be used by another class.

protected give only some strange illusion of restricted access.